### ISSUE TITLE
Where can I find the implementation of eth_sign ?

### ISSUE THREAD

**Issue by joelamouche** [MetaMask/metamask-docs#243]
see comments in https://github.com/MetaMask/metamask-docs/issues/239

I need to access eth_sign without a prefix

- Are all signature functions in the extension coming from https://github.com/MetaMask/eth-sig-util ?
- If that's true, where can I find the eth_sign signature function in that package?

Also this documentation says the message is prefixed as well : https://eth.wiki/json-rpc/API#eth_sign

**Comment by Sdocquir** [comment]
Would love to get an update on this.

**Comment by BboyAkers** [comment]
@Gudahtt any insight on this? 🙂

**Comment by Sdocquir** [comment]
The main issue right now, is that the MM documentation says that eth_sign is not prefixed, but when testing it on my own implementation, it is indeed prefixed. If that's the case, the documentation needs to be changed to reflect that.

**Comment by Gudahtt** [comment]
MetaMask's `eth_sign` method is indeed not prefixed. We offered it before the standard change, so we preserved the old behavior to avoid breaking sites in production.

Our `personal_sign` method is equivalent to the standardized `eth_sign` method.

Our implementation of `personal_sign` can be found here: https://github.com/MetaMask/eth-sig-util/blob/73ace3309bf4b97d901fb66cd61db15eede7afe9/src/personal-sign.ts#L27

Our implementation of `eth_sign` is here: https://github.com/MetaMask/eth-simple-keyring/blob/main/index.js#L79

We do use `eth-sig-util` for most of our signing, but `eth_sign` is directly in the keyring.
